src/Microsoft.ML.OnnxTransformer/OnnxTransform.cs

```diff
@@ -687,6 +723,422 @@ public NamedOnnxValue GetNamedOnnxValue()
                     return OnnxUtils.CreateNamedOnnxValue(_colName, _vBufferDense.GetValues(), _tensorShape);
                 }
             }
+
+            /// <summary>
+            /// <see cref="OnnxTransformer.GetDropColumnsNames"/>
+            /// </summary>
+            public string[] GetDropColumnsNames()
+            {
+                return _parent.GetDropColumnsNames();
+            }
+        }
+
+        /// <summary>
+        /// Similar to <see cref="ColumnBindings"/>, but this class will enable dropping columns from the input
+        /// schema, in order to let OnnxTransformer support the <see cref="ColumnSelectingTransformer"/> onnx export.
+        /// </summary>
+        [BestFriend] // MYTODO: Is this necessary?
+        internal sealed class Bindings // MYTODO: Should I move this inside OnnxDataTransform?
+        {
+            // MYTODO: Should I simply inherit from ColumnBindings, since everything is the same except for the constructor (specifically, only, the way it created the _colMap)?
+
+            // Indices of columns in the merged schema. Old indices are as is, new indices are stored as ~idx.
+            private readonly int[] _colMap;
+
+            /// <summary>
+            /// The indices of added columns in the <see cref="Schema"/>.
+            /// </summary>
+            public IReadOnlyList<int> AddedColumnIndices { get; }
+
+            /// <summary>
+            /// The input schema.
+            /// </summary>
+            public DataViewSchema InputSchema { get; }
+
+            /// <summary>
+            /// The merged schema.
+            /// </summary>
+            public DataViewSchema Schema { get; }
+
+            /// <summary>
+            /// Create a new instance of <see cref="Bindings"/>.
+            /// </summary>
+            /// <param name="input">The input schema that we're adding columns to.</param>
+            /// <param name="dropColumnsNames">Names of the columns to drop, so that they don't propagate from the input schema</param>
+            /// <param name="addedColumns">The columns being added.</param>
+            public Bindings(DataViewSchema input, List<string> dropColumnsNames, DataViewSchema.DetachedColumn[] addedColumns)
+            {
+                Contracts.CheckValue(input, nameof(input));
+                Contracts.CheckValue(addedColumns, nameof(addedColumns));
+
+                InputSchema = input;
+
+                // Construct the indices.
+                // Drop the indicated columns
+                // And drop all hidden columns
+                var indices = new List<int>();
+                var namesUsed = new HashSet<string>();
+                for (int i = 0; i < input.Count; i++)
+                {
+                    if (InputSchema[i].IsHidden || dropColumnsNames.Contains(InputSchema[i].Name)) // MYTODO: Should I drop all hidden columns? Only the ones that are inside the dropColumnsNames list?
+                        continue;
```

Comment:
The `ColumnSelectingTransformer` has a couple of options such as if it should drop Hidden columns or not, and if it should do it only in the columns it's dropping or not. These options are lost when exporting the model to onnx, so there's no way to know if when applying the onnx model what should we do with Hidden columns.

So I simply decided to drop _all_ the Hidden columns (regardless if its name was in the input or output of the onnx model). I think it's better for the user, since it might be even more confusing to see some non-hidden columns being dropped, while the hidden ones remain.